lib/Backends/CPU/libjit/libjit.cpp

```diff
@@ -1531,6 +1532,13 @@ libjit_dump_tensor(uint8_t *tensor, size_t *tensorDim, size_t numDimsTensor,
   }
 }
 
+void libjit_write_timestamp(uint64_t *tensor, size_t offset) {
+  uint64_t ts = std::chrono::duration_cast<std::chrono::milliseconds>(
```

Comment:
In general, we try to avoid calling any C++ functions from the `libjit` to avoid any possible issues with C++ symbols resolution at run-time. If it is possible to replace it by an equivalent `libc`-call, it would be nice.

Reply:
OK I can use gettimeofday here instead.

Reply:
I'm coming back on this, gettimeofday is super bad (e.g. https://blog.habets.se/2010/09/gettimeofday-should-never-be-used-to-measure-time.html) and there isn't a portable alternative that's not c++.

How serious is using C++ in the JIT? This could get very complicated.

Reply:
@nickgg :-( Hmm, if there is no portable way other than C++, we could use `std::chrono` as you suggest, provided that all tests pass including the ones that use JIT compilation and AOT compilation. But add a comment before this `chrono` that we fallback to C++ here because there is no portable C solution.

If people then have issues with this because e.g. C++ symbols cannot be found at runtime, they should let us know and we may reconsider what to do here.

Reply:
For now I'll add that comment and come back to it.

We can use clock_gettime on Linux and OSX >= 12.12, and fall back to something else on other platforms. gettimeofday might be acceptable there? I filed #2397 for this as a follow up.